packages/flutter/lib/src/rendering/editable_line.dart

```diff
@@ -177,14 +198,41 @@ class RenderEditableLine extends RenderBox {
     _lastTapDownPosition = null;
     if (onSelectionChanged != null) {
       TextPosition position = _textPainter.getPositionForOffset(globalToLocal(global).toOffset());
-      onSelectionChanged(new TextSelection.fromPosition(position));
+      onSelectionChanged(new TextSelection.fromPosition(position), this);
     }
   }
 
   void _handleTapCancel() {
+    // longPress arrives after tapCancel, so remember the tap position.
+    _longPressPosition = _lastTapDownPosition;
     _lastTapDownPosition = null;
   }
 
+  void _handleLongPress() {
+    final Point global = _longPressPosition;
+    _longPressPosition = null;
+    if (onSelectionChanged != null) {
+      TextPosition position = _textPainter.getPositionForOffset(globalToLocal(global).toOffset());
+      onSelectionChanged(_selectWordAtOffset(position), this);
+    }
+  }
+
+  TextSelection _selectWordAtOffset(TextPosition position) {
+    // TODO(mpcomplete): Placeholder. This is surely the wrong way to do this
+    // with UTF-8. Also this doesn't work when the text is scrolled. Bleh.
+    String str = text.toPlainText();
+    int start = position.offset - 1;
+    while (start >= 0 && str[start] != ' ')
+      --start;
+    ++start;
```

Comment:
Yeah, this only works in English (and not even really then).  You should ask the engine to compute the word boundaries for you.  It's a super complex process in some languages.  In C++, there's a WordIterator that knows how to do it.